compiler-core/src/type_/expression.rs

```diff
@@ -707,6 +722,40 @@ impl<'a, 'b> ExprTyper<'a, 'b> {
                 .map_err(|e| e.case_clause_mismatch().into_error(typed_clause.location()))?;
             typed_clauses.push(typed_clause);
         }
+
+        // Because exhaustiveness checking in presence of multiple subjects is similar
+        // to full exhaustiveness checking of tuples or other nested record patterns,
+        // and we currently only do only limited exhaustiveness checking of custom types
+        // at the top level of patterns, only consider case expressions with one subject.
+        if subjects_count == 1 {
```

Comment:
Would it be clearer if this code is extracted into a named function?

Could then convert this check + nesting into a guard then.

```rust
if subjects_count != 1 {
  return Ok(());
}

// rest here...
```

Reply:
Updated

Reply:
This was done in https://github.com/gleam-lang/gleam/pull/1462/commits/fe9534deaf0f186ecc27e13223be62a7f939be80